### ISSUE TITLE
apt-get install endaga-osmocom fails

### ISSUE THREAD

**Issue by sharma-sagar** [facebookincubator/CommunityCellularManager#89]
While trying to setup osmocom client vagrant up osmocom successfully executed but after,
**vagrant ssh osmocom** and **sudo apt-get update** the below command fails:

**apt-get install endaga-osmocom**
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 **endaga-osmocom : Depends: openggsn (>= 0.92) but it is not installable
                  Depends: osmocom-sgsn (>= 0.15.0) but it is not installable
                  Depends: freeswitch-meta-vanilla but it is not going to be installed**
E: Unable to correct problems, you have held broken packages.

**Comment by matt9j** [comment]
I'm running into this problem as well, and believe it stems from the migrations done at osmocom to split up osmo-nitb and their recent repository reogranization. (some information here: https://osmocom.org/projects/cellular-infrastructure/wiki/Osmocom_Network_In_The_Box).

**Comment by matt9j** [comment]
I've incorporated the version update suggested by epruves in https://github.com/pcarivbts/CommunityCellularManager/issues/8 in a working branch here: https://github.com/uw-ictd/CommunityCellularManager/tree/devel-client-maintenance, along with some other fixes and cleanup. I'm running into a second order issue though: 

```
vagrant@endaga-client-osmocom:~$ sudo apt-get install endaga-osmocom
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 endaga-osmocom : Depends: freeswitch-mod-smpp but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
```

Attempting to install freeswitch-mod-smpp:
```
vagrant@endaga-client-osmocom:~$ sudo apt-get install freeswitch-mod-smpp
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 freeswitch-mod-smpp : Depends: libsmpp0 but it is not installable
E: Unable to correct problems, you have held broken packages.
```

As far as I can find, I cannot locate any kind of package for libsmpp0. Osmocom maintains the source package libsmpp34 and a distribution package libsmpp1: http://download.opensuse.org/repositories/network:/osmocom:/latest/Debian_8.0/amd64/ 

```
vagrant@endaga-client-osmocom:~$ sudo apt-cache search libsmpp
libsmpp1 - Open PDU SMPP packaging and unpackaging tool
libsmpp34-dbg - Debug symbols for libsmpp34
libsmpp34-dev - Development files for libsmpp34
libsmpp34-utils - Utilities for SMPP
libsmpp34-0 - Open PDU SMPP packaging and unpackaging tool
```

I tried to install libsmpp1 and manually install the freeswitch_mod_smpp deb ignoring dependencies, but doing so results in freeswitch crashing shortly after startup (and leads to issues doing apt-get upgrade). Manually installing libsmpp34-0 appears to work okay, but I don't have a test setup working right now to test end-end. Freeswitch does appear to start without crashing though.

From my understanding of the resolution to #28 , I believe to address the issue either an ABI compatible libsmpp0 package should be added to the etagecom repo, the freeswitch_mod_smpp package should be updated to point to libsmpp34-0, or the provided version of freeswitch mod smpp should be recompiled against libsmpp1.

**Comment by shaddi** [comment]
Yeah, I think that sounds about right. Did you try re-building FS and `freeswitch_mod_smpp` from source already to test that fix?

**Comment by matt9j** [comment]
> Did you try re-building FS and freeswitch_mod_smpp from source already to test that fix?

I attempted to, but was orthogonally having some build issues. I found instructions for building FS here: https://gitlab.com/marc.ferland/meta-nrw-ccm and in the regular freeswitch docs: https://freeswitch.org/confluence/display/FREESWITCH/Debian+8+Jessie

Both suggest essentially checking out the target tag, and then running `./debian/util.sh build-all -aamd64 -cjessie`. I never got any generated .deb files though... it seems the only output generated files were
```
freeswitch_1.6.12+git~20180525T192404Z~b91a0a6f2b-1~jessie+1.debian.tar.xz                   
freeswitch_1.6.12+git~20180525T192404Z~b91a0a6f2b-1~jessie+1.dsc                             
freeswitch_1.6.12+git~20180525T192404Z~b91a0a6f2b-1~jessie+1_source.changes                  
freeswitch_1.6.12+git~20180525T192404Z~b91a0a6f2b.orig.tar.xz
```
I'm probably missing something here, but was doing this late yesterday and never got it figured out.

**Comment by matt9j** [comment]
Here's the vagrant info I used to setup my build VM: https://github.com/uw-ictd/freeswitch-env

**Comment by matt9j** [comment]
Also, here's the commit that made the change: https://github.com/osmocom/libsmpp34/commit/6996e5dba9603208609f4604a3b3d59f88c85d90
